vue/components/ui/molecules/button-dropdown/button-dropdown.vue

```diff
@@ -0,0 +1,169 @@
+<template>
+    <div class="button-dropdown">
+        <div class="button button-principal" v-on:click.stop="primaryButtonClicked">

```

Comment:
Maybe use the logic onPrimaryButtonClicked, the same with onToggleDropdown and onDropdownItemClicked.